src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,40 @@
+package duke.task;
+
+import duke.DateUtil;
+import duke.Duke;
+import duke.DukeException;
+
+import java.util.Date;
+
+public class Event extends Task {
+
+    private Date at;
+
+    /**
+     * Constructs an event with the given description and date/time.
+     *
+     * @param description the description of the event.
+     * @param at the date/time of the event.
+     * @throws DukeException if the event has no date/time.
+     */
+    public Event(String description, Date at) throws DukeException {
+        super(description);
+        if (at.equals("")) {
+            throw new DukeException("The date/time of " + this.getTypeNameWithQuantifier() + " cannot be empty.");
```

Comment:
getTypeNameWithQuantifier() seems to work alright,.getClass.toLowerCase could perhaps be an alternative workaround